adev/shared-docs/components/viewers/docs-viewer/docs-viewer.component.spec.ts

```diff
@@ -208,4 +216,41 @@ describe('DocViewer', () => {
 
     expect(renderComponentSpy).not.toHaveBeenCalled();
   });
+
+  it('should setup copy link functionality for docs-anchor elements', async () => {
+    const fixture = TestBed.createComponent(DocViewer);
+    fixture.componentRef.setInput('docContent', exampleContentWithDocsAnchor);
+
+    fixture.detectChanges();
+    await fixture.whenStable();
+
+    const copyLinkButton = fixture.debugElement.query(By.directive(CopyLinkButton));
+    expect(copyLinkButton).toBeTruthy();
+
+    const anchor = fixture.nativeElement.querySelector('a.docs-anchor') as HTMLAnchorElement;
+    expect(anchor).toBeTruthy();
+
+    const copyButton = fixture.nativeElement.querySelector(
+      'docs-copy-link-button',
+    ) as HTMLButtonElement;
+    expect(copyButton).toBeTruthy();
+  });
+
+  it('should copy link to clipboard when copy button is clicked', async () => {
+    const clipboard = TestBed.inject(Clipboard);
+    const clipboardSpy = spyOn(clipboard, 'copy').and.returnValue(true);
+
+    const fixture = TestBed.createComponent(DocViewer);
+    fixture.componentRef.setInput('docContent', exampleContentWithDocsAnchor);
+
+    fixture.detectChanges();
+    await fixture.whenStable();
+
+    const copyButton = fixture.nativeElement.querySelector(
+      'docs-copy-link-button',
+    ) as HTMLButtonElement;
```

Comment:
```suggestion
    );
```
Type assertion looks unnecesary